lib/daru/maths/statistics/vector.rb

```diff
@@ -75,19 +71,70 @@ def factors
         # @param return_type [Symbol] Data type of the returned value. Defaults
         #   to returning only the maximum number but passing *:vector* will return
         #   a Daru::Vector with the index of the corresponding maximum value.
-        def max return_type=:stored_type
-          max_value = @data.max
-          if return_type == :vector
-            Daru::Vector.new({index_of(max_value) => max_value}, name: @name, dtype: @dtype)
+        def max
```

Comment:
@zverok : I had moved this functionality into `Daru::Vector#max_index` previously used to call `Daru::Vector#max` with `return_type` argument. And as it (`Daru::Vector#max_index`) is currently being used only from `Daru::DataFrame#max`, I thought it would be better to have this as `Daru::Vector#max_index` function - so that `Daru::Vector#max` could handle this with blocks :

```
dv = Daru::Vector.new ['a', 'bb', 'ccc'], index: [:a, :b, :c]
=> #<Daru::Vector(3)>
   a   a
   b  bb
   c ccc
dv.index_of_max { |i| i.size }
=> :c 
```